### DISCUSSION TITLE
WalletConnect Guide via API

### DISCUSSION THREAD

**Discussion by johnpaulcas** [WalletConnect/walletconnect-monorepo#2526] [Developer Support]
Hello, I'm seeking assistance regarding WalletConnect's API implementation. I've noticed that their documentation largely emphasizes UI interactions. However, our aim is to establish a connection through the API, where we plan to send a URI to the user that can be scanned as a QR code. We're also looking to have an API that can return a session status to verify if the user is connected. Any advice or procedural guidance for this would be greatly appreciated. Thank you.

**Comment by finessevanes** [comment]
It sounds like you may want to use our Sign API: https://docs.walletconnect.com/2.0/api/sign

**Reply by johnpaulcas** [reply]
Hi @finessevanes upon checking on the docs you sent, it seems is still pointing to some UI implementation but nothing with regards or steps on how to do it via API? may I ask if you have docs on how to do the wallet connection, sign transactions on via API so we can use it to other protocol like unreal, thanks!

**Reply by glitch-txs** [reply]
You can use the display_uri event, it will give you the Uri to create your own QRCode. You only need to remove the @walletconnect/modal from the example
Ethereum-provider also supports this and it's EIP-1193 compliant

For something more low level I'd check the Core API

**Reply by johnpaulcas** [reply]
Cool! @glitch-txs @finessevanes I was be able to connect to my wallet, so it does mean also we need to handle the session manually on the backend right? but may we ask how to determine on which wallet is connection? and some docs to like for example interact on smart contract via API? on version 1 we need to populate the provider, is this same with version 2? any help will greatly appreciated

Thanks!

**Reply by johnpaulcas** [reply]
hi @glitch-txs just want to followup something, so does wallet connect handle the provider when interacting with a smart contract using `signClient.request` with request method `eth_sendTransaction`?

**Reply by glitch-txs** [reply]
Transactions are handled by the wallet's RPC endpoints, reading calls by the WalletConnect's provider if I'm not mistaken

**Reply by johnpaulcas** [reply]
It just confuse me, so I generated qr code, user successfully scan the qrcode and we get the session and save to database, when user -> interact to smart contract, we call `.request` on our backend with `method` `eth_sendTransaction`, so this will popup to user wallet right? so is that mean that the wallet for example metamask will automatically provide the provider for that?

**Reply by glitch-txs** [reply]
That's correct, you are actually communicating with the Relay and the Relay with the wallet.
I recommend taking a look at the specs
https://docs.walletconnect.com/2.0/specs/

**Reply by johnpaulcas** [reply]
thanks @glitch-txs, guess this is not the case before on v1 where you still need to pass the provider that's why its confuse me
